### PR TITLE
Added remove_texts on base, elasticsearch, pgvector vectorsearch

### PR DESCRIPTION
**Pull request by gregogalante**

### CODE REVIEW

**Comment by gregogalante** [comment]
@andreibondarev I have a little trouble understanding the problem  with pgvector remove_texts. Can you check what the result value on tests?

**Comment by andreibondarev** [comment]
> @andreibondarev I have a little trouble understanding the problem with pgvector remove_texts. Can you check what the result value on tests?

Hmmm... is it because we're adding other records to pgvector before this `remove_texts` spec is executed? Odd.

**Comment by andreibondarev** [comment]
@gregogalante I cherry-picked some of your commits out to focus on this PR on just the Pgvector.

File: spec/langchain/vectorsearch/pgvector_spec.rb [hunk 1]
```diff
@@ -98,6 +98,36 @@
...
+          })
+      end
+
+      it "removes texts" do
+        values = subject.add_texts(texts: ["Hello World", "Hello World"])
+        ids = values.flatten
+        expect(ids.length).to eq(2)
+
+        result = subject.remove_texts(ids: ids)
+        expect(result.size).to eq(2)
```
[the remark this answers is not in the document]
Comment (gregogalante) [reply spec/langchain/vectorsearch/pgvector_spec.rb:127]: @andreibondarev I don't have a local pg vector database to test it. Can you try to run only remove_texts test to check length?

[the remark this answers is not in the document]
Comment (andreibondarev) [reply spec/langchain/vectorsearch/pgvector_spec.rb:127]: @gregogalante Turns out that `result` is the actual number of elements deleted:
```ruby
        result = subject.remove_texts(ids: ids)
        expect(result.size).to eq(2)
```

But when you call `result.size` it's the same as calling `2.size` which is the number of bytes that Integer represent.

**Pull request merged**